### PR TITLE
Added 'wrap-type' option for sentence-wrapping

### PR DESCRIPTION
**Pull request by tbazant**
+ removed default slicing text nodes into sentences with regexp as it collide with `wrap-length`
+ introduced a new `wrap-type` option which defaults to `length` (old behavior) and introduces `sentence` value to break lines at sentence break instead of line length
- sadly, the Ruby version is not updated, i cant do Ruby

### CODE REVIEW

**Review by someth2say — CHANGES_REQUESTED** [review]
This is amazing! Thanks a lot!
I would propose, for the sake of retro-compatibility, to default the `wrap-type` to `sentence`.

With your consent, I would also take some time to port your changes to Ruby (I can barely do Ruby, but I'll give it a try).

**Comment by tbazant** [comment]
> This is amazing! Thanks a lot!
you're welcome
> I would propose, for the sake of retro-compatibility, to default the `wrap-type` to `sentence`.
of course, make sentence the default
> 
> With your consent, I would also take some time to port your changes to Ruby (I can barely do Ruby, but I'll give it a try).
i agree on both of your points, go ahead :-)

File: bin/xmlformat.pl [hunk 1]
```diff
@@ -176,6 +174,7 @@ sub new
     "format"    => "block",
     "normalize"   => "no",
     "subindent"   => 0,
+    "wrap-type" => "length",
```
Comment (tbazant) [inline bin/xmlformat.pl]:
```suggestion
    "wrap-type" => "sentence",
```

File: bin/xmlformat.pl [hunk 2]
```diff
@@ -189,6 +188,7 @@ sub new
     "format"    => "block",
     "normalize"   => "no",
     "subindent"   => 1,
+    "wrap-type" => "length",
```
Comment (tbazant) [inline bin/xmlformat.pl]:
```suggestion
    "wrap-type" => "sentence",
```

**Review by tbazant — COMMENTED** [review]
Make wrap type 'sentence' the default

**Comment by tbazant** [comment]
@someth2say how does it look with the ruby version of xmlformat ? Do we need the ruby version at all if we have perl one?

**Comment by someth2say** [comment]
Hi!
I should say yes, the ruby version is mandatory, as many people is using it, and we need consistency.
I hope you don't mind I pushed to your branch the implementation and some tests. 
If you are ok with it, I can merge it and build the container image and cook a release.

**Review by someth2say — APPROVED** [review]

**Review by tbazant — COMMENTED** [review]
looks good, thanks a lot!

**Comment by someth2say** [comment]
@all-contributors add @tbazant for code and design

**Comment by tbazant** [comment]
thanks for adding me there :-)

**Comment by someth2say** [comment]
Absolutely my pleasure!

**Pull request merged**